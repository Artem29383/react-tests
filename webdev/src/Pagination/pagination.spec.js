import React from "react";
import Pagination from "./Pagination";

const props = {
  lastPage: 20,
}

const setUp = () => shallow(<Pagination />);
const setUpProps = () => shallow(<Pagination {...props} />);
const setUpWithProps = () => shallow(<Pagination  {...props} lastPage={20} />);


describe('Posts component', () => {
  let component;
  let componentWithProps;
  let instance;
  let instanceWithProps;

  beforeEach(() => {
    component = setUp();
    componentWithProps = setUpProps();
    instance = component.instance();
    instanceWithProps = componentWithProps.instance();
  })


  it('Render pagination component without Props', () => {
    expect(component).toMatchSnapshot();
  })

  it('Render pagination component with Props', () => {
    expect(componentWithProps).toMatchSnapshot();
  })

  it('Render pagination component for last pages', () => {
    const actualComponent = setUpWithProps({ page: 15 });
    expect(actualComponent).toMatchSnapshot();
  })

  it('Render pagination component without three dots in the middle', () => {
    const actualComponent = setUpWithProps({ page: 16 });
    expect(actualComponent).toMatchSnapshot();
  })

  it('Render pagination component with three dots and three buttons in the end', () => {
    const actualComponent = setUpWithProps({ page: 19 });
    expect(actualComponent).toMatchSnapshot();
  })


  describe('defaultProps', () => {
    it('should use default change', function () {
      const result = Pagination.defaultProps.onClick();
      expect(result).toBe(undefined);
    });
  })
})