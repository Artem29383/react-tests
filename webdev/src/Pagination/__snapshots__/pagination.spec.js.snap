// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component Render pagination component for last pages 1`] = `
<div
  className="paginationWrapper"
>
  <button
    className="active"
    data-name={0}
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className=""
    data-name={1}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    data-name={2}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Posts component Render pagination component with Props 1`] = `
<div
  className="paginationWrapper"
>
  <button
    className="active"
    data-name={0}
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className=""
    data-name={1}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    data-name={2}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Posts component Render pagination component with three dots and three buttons in the end 1`] = `
<div
  className="paginationWrapper"
>
  <button
    className="active"
    data-name={0}
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className=""
    data-name={1}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    data-name={2}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;

exports[`Posts component Render pagination component without Props 1`] = `
<div
  className="paginationWrapper"
/>
`;

exports[`Posts component Render pagination component without three dots in the middle 1`] = `
<div
  className="paginationWrapper"
>
  <button
    className="active"
    data-name={0}
    key="0"
    onClick={[Function]}
  >
    0
  </button>
  <button
    className=""
    data-name={1}
    key="1"
    onClick={[Function]}
  >
    1
  </button>
  <button
    className=""
    data-name={2}
    key="2"
    onClick={[Function]}
  >
    2
  </button>
  ...
  <button
    className=""
    data-name={17}
    key="17"
    onClick={[Function]}
  >
    17
  </button>
  <button
    className=""
    data-name={18}
    key="18"
    onClick={[Function]}
  >
    18
  </button>
  <button
    className=""
    data-name={19}
    key="19"
    onClick={[Function]}
  >
    19
  </button>
  <button
    data-name="next"
    onClick={[Function]}
  >
    &gt;&gt;
  </button>
</div>
`;
